### PR TITLE
Fix for flaky upgrade test

### PR DESCRIPTION
**Pull request by pravinpushkar**
Signed-off-by: Pravin Pushkar <ppushkar@microsoft.com>

# Description

Fix for flaky upgrade related tests

## Issue reference

Please reference the issue this PR will close: #794 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [NA] Created/updated tests
* [NA] Extended the documentation

### CODE REVIEW

File: tests/e2e/common/common.go [hunk 1]
```diff
@@ -68,7 +72,7 @@ type TestCase struct {
 	Callable func(*testing.T)
 }
 
-func UpgradeTest(details VersionDetails) func(t *testing.T) {
+func UpgradeTest(details VersionDetails, opts TestOptions) func(t *testing.T) {
```
Comment (surenderssm) [inline tests/e2e/common/common.go:75]: nit : how about opts -> testOptions?

File: tests/e2e/common/common.go [hunk 2]
```diff
@@ -665,14 +668,24 @@ func waitAllPodsRunning(t *testing.T, namespace string, done, podsRunning chan s
...
+				if size == 0 {
+					countOfReadyPods++
+				}
 			}
 		}
-		if len(list.Items) == count {
+		if haEnabled && len(list.Items) == countOfReadyPods && countOfReadyPods == numberOfExpectedPodsHAMode {
+			podsRunning <- struct{}{}
+		} else if !haEnabled && len(list.Items) == countOfReadyPods && countOfReadyPods == numberOfExpectedPodsNonHAMode {
 			podsRunning <- struct{}{}
```
Comment (surenderssm) [inline tests/e2e/common/common.go:689]: can both of these conditions moved to OR, as they end up having the same affect `podsRunning <- struct{}{}`

File: tests/e2e/common/common.go (hunk 1 above)
Comment (shubham1172) [reply tests/e2e/common/common.go:75]: Go prefers smaller names.
https://github.com/golang/go/wiki/CodeReviewComments#variable-names

File: tests/e2e/common/common.go [hunk 3]
```diff
@@ -665,14 +668,24 @@ func waitAllPodsRunning(t *testing.T, namespace string, done, podsRunning chan s
...
+				}
+				if size == 0 {
+					countOfReadyPods++
+				}
 			}
 		}
-		if len(list.Items) == count {
+		if haEnabled && len(list.Items) == countOfReadyPods && countOfReadyPods == numberOfExpectedPodsHAMode {
+			podsRunning <- struct{}{}
+		} else if !haEnabled && len(list.Items) == countOfReadyPods && countOfReadyPods == numberOfExpectedPodsNonHAMode {
```
Comment (mukundansundar) [inline tests/e2e/common/common.go:688]:
```suggestion
		if len(list.Items) == countOfReadyPods 
		&& (
		  ( haEnabled && countOfReadyPods == numberOfExpectedPodsHAMode ) 
		  || ( !haEnabled && countOfReadyPods == numberOfExpectedPodsNonHAMode )
		) {
			podsRunning <- struct{}{}
		} 
```

Comment (mukundansundar) [reply tests/e2e/common/common.go:688]: can change `numberOfExpectedPodsHAMode` to `numHAPods` and 
`numberOfExpectedPodsNonHAMode` to `numNonHAPods`

**Review by mukundansundar — COMMENTED** [review]
lgtm

File: tests/e2e/common/common.go [hunk 4]
```diff
@@ -244,7 +247,7 @@ func StatusTestOnInstallUpgrade(details VersionDetails, opts TestOptions) func(t
 				"dapr-operator":         {details.RuntimeVersion, "3"},
 			}
 		}
-
+		t.Log(output)
```
Comment (mukundansundar) [inline tests/e2e/common/common.go:250]: is this needed ?

File: tests/e2e/common/common.go [hunk 5]
```diff
@@ -46,6 +46,10 @@ const (
 )
 
 const DaprTestNamespace = "dapr-cli-tests"
+const (
+	numHAPods    = 13
+	numNonHAPods = 5
+)
```
Comment (mukundansundar) [inline tests/e2e/common/common.go:52]:
```suggestion
const (
        DaprTestNamespace = "dapr-cli-tests"
	numHAPods               = 13
	numNonHAPods        = 5
)
```

**Review by mukundansundar — COMMENTED** [review]
small changes ...

File: tests/e2e/common/common.go (hunk 4 above)
Comment (pravinpushkar) [reply tests/e2e/common/common.go:250]: Not really, just added to get actual status.

Comment (mukundansundar) [reply tests/e2e/common/common.go:250]: please remove ...

**Review by mukundansundar — APPROVED** [review]
lgtm.

**Pull request merged**